client_taxi_driver/shared/src/commonMain/kotlin/presentation/map/MapScreen.kt

```diff
@@ -0,0 +1,292 @@
+import org.jetbrains.compose.resources.painterResource
+import presentation.base.BaseScreen
+import presentation.resources.Resources
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalResourceApi::class)
+class MapScreen :
+    BaseScreen<MapScreenModel, MapScreenUiState, MapUiEffect, MapScreenInteractionListener>() {
+
+    @Composable
+    override fun onRender(state: MapScreenUiState, listener: MapScreenInteractionListener) {
+        Box(
+            modifier = Modifier.fillMaxSize()
+        ) {
+            BpAppBar(
+                isBackIconVisible = false,
+                title = "${Resources.strings.mapScreenAppBarTitle}${state.userName}"
+            ) {
+                Box(
+                    modifier = Modifier
+                        .padding(end = 16.dp)
+                        .width(40.dp)
+                        .height(40.dp)
+                        .background(
+                            color = Theme.colors.hover,
+                            shape = RoundedCornerShape(size = 8.dp)
+                        ).noRippleEffect(onClick = listener::onClickCloseIcon),
+                ) {
+                    Icon(
+                        modifier = Modifier.align(Alignment.Center),
+                        painter = painterResource(Resources.images.close),
+                        contentDescription = Resources.strings.close,
+                        tint = Theme.colors.contentPrimary
+                    )
+                }
+            }
+
+            AnimatedVisibility(
+                modifier = Modifier.align(Alignment.BottomCenter),
+                visible = state.isLoading,
+                enter = slideInVertically { it } + fadeIn(),
+                exit = slideOutVertically { it } + fadeOut()
+            ) {
+                FindingRideCard()
+            }
+
+            AnimatedVisibility(
+                modifier = Modifier.align(Alignment.BottomCenter),
+                visible = state.isNewOrderFound,
+                enter = slideInVertically { it } + fadeIn(),
+                exit = slideOutVertically { it } + fadeOut()
+            ) {
+                NewOrderCard(
+                    modifier = Modifier.align(
+                        Alignment.BottomCenter,
+                    ),
+                    state = state.orderInfoUiState,
+                    listener = listener,
+                )
+            }
+
+            AnimatedVisibility(
+                modifier = Modifier.align(Alignment.BottomCenter),
+                visible = state.isAcceptedOrder,
+                enter = slideInVertically { it } + fadeIn(),
+                exit = slideOutVertically { it } + fadeOut()
+            ) {
+                OrderCard(
+                    modifier = Modifier.align(
+                        Alignment.BottomCenter,
+                    ),
+                    state = state.orderInfoUiState,
+                    listener = listener,
+                )
+            }
+        }
+    }
```

Comment:
this is a duplication code in the animatedVisibility i think we can put it in a private function to reduce it, give it a try and let's see how it will be

Reply:
Done